File: apps/ios/Packages/MarkerDomain/Sources/MarkerDomain/MarkerDomainModule.swift
public enum MarkerDomainBoundary: String, CaseIterable, Sendable {
    case trackerRepository = "TrackerRepository"
    case trackingEntryRepository = "TrackingEntryRepository"
    case statisticsRepository = "StatisticsRepository"
    case trackerReminderScheduler = "TrackerReminderScheduler"
}

public protocol TrackerRepository: Sendable {}

public protocol TrackingEntryRepository: Sendable {}

public protocol StatisticsRepository: Sendable {}

public protocol TrackerReminderScheduler: Sendable {}

@available(*, deprecated, renamed: "TrackingEntryRepository")
public typealias CheckInRepository = TrackingEntryRepository

@available(*, deprecated, renamed: "TrackerReminderScheduler")
public typealias ReminderScheduler = TrackerReminderScheduler

public enum MarkerDomainModule {
    public static let defaultBoundaries: [MarkerDomainBoundary] = MarkerDomainBoundary.allCases
}

File: apps/android/domain/src/main/kotlin/com/edge/marker/domain/MarkerDomainModule.kt
package com.edge.marker.domain

enum class MarkerDomainBoundary(val rawValue: String) {
    TRACKER_REPOSITORY(rawValue = "TrackerRepository"),
    TRACKING_ENTRY_REPOSITORY(rawValue = "TrackingEntryRepository"),
    STATISTICS_REPOSITORY(rawValue = "StatisticsRepository"),
    TRACKER_REMINDER_SCHEDULER(rawValue = "TrackerReminderScheduler"),
}

interface TrackerRepository

interface TrackingEntryRepository

interface StatisticsRepository

interface TrackerReminderScheduler

@Deprecated(message = "Use TrackerRepository instead", replaceWith = ReplaceWith("TrackerRepository"))
typealias HabitRepository = TrackerRepository

@Deprecated(message = "Use TrackingEntryRepository instead", replaceWith = ReplaceWith("TrackingEntryRepository"))
typealias CheckInRepository = TrackingEntryRepository

@Deprecated(message = "Use TrackerReminderScheduler instead", replaceWith = ReplaceWith("TrackerReminderScheduler"))
typealias ReminderScheduler = TrackerReminderScheduler

object MarkerDomainModule {
    val defaultBoundaries: List<MarkerDomainBoundary> = MarkerDomainBoundary.entries.toList()
}
